src/core/mac/mac.cpp

```diff
@@ -1067,6 +1074,52 @@ void Mac::TransmitDoneTask(otRadioFrame *aFrame, otRadioFrame *aAckFrame, otErro
 
 #endif // OPENTHREAD_CONFIG_LEGACY_TRANSMIT_DONE
 
+otError Mac::RadioTransmit(Frame *aSendFrame)
+{
+#if OPENTHREAD_CONFIG_STAY_AWAKE_BETWEEN_FRAGMENTS
+    // Cancel delay sleep timer
+    mReceiveTimer.Stop();
+#endif
+    // Transmit packet
+    return otPlatRadioTransmit(GetInstance(), static_cast<otRadioFrame *>(aSendFrame));
+}
+
+otError Mac::RadioReceive(uint8_t aChannel)
+{
+#if OPENTHREAD_CONFIG_STAY_AWAKE_BETWEEN_FRAGMENTS
+    // Cancel delay sleep timer
+    mReceiveTimer.Stop();
+#endif
+    // Receive
+    return otPlatRadioReceive(GetInstance(), aChannel);
+}
+
+void Mac::RadioSleep(void)
+{
+#if OPENTHREAD_CONFIG_STAY_AWAKE_BETWEEN_FRAGMENTS
+
+    if (GetFramePending())
+    {
+        // Restart delay sleep timer
+        mReceiveTimer.Start(kDataPollTimeout);
```

Comment:
I suggest that we define and use a new constant (say `kSleepDelay`) here. It gives us flexibility to change the delay or poll timeout independent of each other.